File: app/Models/PlatformInfo.php
<?php

namespace Irail\Models;

class PlatformInfo
{
    private string $id;
    private string $designation;
    private bool $hasChanged;

    /**
     * @param string $parentStopId
     * @param string $designation
     * @param bool   $hasChanged
     */
    public function __construct(string $parentStopId, string $designation, bool $hasChanged)
    {
        $this->id = ($parentStopId && $designation) ? $parentStopId . "#" . $designation : null;
        $this->designation = $designation;
        $this->hasChanged = $hasChanged;
    }

    /**
     * @return string|null
     */
    public function getId(): string
    {
        return $this->id;
    }

    /**
     * @return string|null
     */
    public function getDesignation(): string
    {
        return $this->designation;
    }

    /**
     * @return bool
     */
    public function hasChanged(): bool
    {
        return $this->hasChanged;
    }

}

File: src/main/java/be/irail/api/dto/PlatformInfo.java
package be.irail.api.dto;

/**
 * Represents information about a station platform.
 * Includes the platform designation and whether it has changed from the originally scheduled one.
 */
public class PlatformInfo {
    private final String id;
    private final String designation;
    private final boolean hasChanged;

    /**
     * Constructs a new PlatformInfo.
     *
     * @param parentStopId the identifier of the parent station stop
     * @param designation the platform designation (e.g., "4", "A")
     * @param hasChanged true if the platform has changed from the original schedule
     */
    public PlatformInfo(String parentStopId, String designation, boolean hasChanged) {
        this.id = (parentStopId != null && designation != null) ? parentStopId + "#" + designation : null;
        this.designation = designation != null ? designation : "?";
        this.hasChanged = hasChanged;
    }

    /**
     * Gets the unique identifier for this platform at this station.
     * @return the platform ID
     */
    public String getId() {
        return this.id;
    }

    /**
     * Gets the platform designation (e.g., name or number).
     * @return the platform designation
     */
    public String getDesignation() {
        return this.designation;
    }

    /**
     * Checks if the platform has changed from the originally scheduled one.
     * @return true if the platform has changed
     */
    public boolean hasChanged() {
        return this.hasChanged;
    }
}
